### PR TITLE
feat: event based history tables, dbs and importer

### PR DESCRIPTION
**Pull request by tmrdlt**
- Added historyV2 table & db
- Added historyErrorV2 table & db:
  - Remove `totalSupply`
  - Use `BigInt` for all numbers
- Added importerV2
- Fix: Re-added `TasksModule` to `AppModule`
- Fixed udpate in `PairLiquidityInfoHistoryDb.upsert`
- Fixed `ClientModule`

### CODE REVIEW

File: prisma/migrations/20240404105443_history_v2/migration.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "microBlockTime" BIGINT NOT NULL,
+    "logIndex" INTEGER NOT NULL,
+    "microBlockHash" TEXT NOT NULL,
+    "transactionHash" TEXT NOT NULL,
+    "totalSupply" TEXT,
+    "reserve0" TEXT,
+    "reserve1" TEXT,
+    "deltaReserve0" TEXT,
+    "deltaReserve1" TEXT,
+    "fiatPrice" TEXT,
```
Comment (kenodressel) [inline prisma/migrations/20240404105443_history_v2/migration.sql]: maybe we can bring the changes discussed for #21 already in this migration

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        }
+        const lastSyncedHeight = lastSyncedLog?.height || 0;
+        const lastSyncedBlockTime = lastSyncedLog?.microBlockTime || 0n;
+        const lastSyncedLogIndex = lastSyncedLog?.logIndex || -1;
+
+        // Determine which logs to sync based on the lastly synced log
+        // Strategy:
+        // 1. Always (re-)fetch everything within the 10 most recent key blocks (currentHeight - 10).
+        // 2. If the history is outdated, fetch everything since the lastly synced log
+        const isHistoryOutdated = lastSyncedHeight < currentHeight - 10;
```
Comment (kenodressel) [inline src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer-v2.service.ts]: lets move the `10` to a constant similar to the hours

File: src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer.service.ts [hunk 3]
```diff
@@ -1,19 +1,19 @@
 import { Injectable, Logger } from '@nestjs/common';
-import { MdwHttpClientService } from '../clients/mdw-http-client.service';
+import { MdwHttpClientService } from '../../clients/mdw-http-client.service';
```
Comment (kenodressel) [inline src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer.service.ts:2]: should we introduce some sort of absolute paths?

https://stackoverflow.com/questions/57439410/how-nestjs-configures-path-aliases-in-a-project

Comment (tmrdlt) [reply src/tasks/pair-liquidity-info-history-importer/pair-liquidity-info-history-importer.service.ts:2]: Good idea, I'll introduce this in a different PR

**Review by kenodressel — APPROVED** [review]

**Pull request merged**